### ISSUE TITLE
LSP: show workspace/buffer diagnostics in the quickfix window

### ISSUE THREAD

**Issue by urandom** [neovim/neovim#14481]
Would it be possible to have 2 functions in `vim.lsp.diagnostic` for populating the quickfix window with the buffer or workspace diagnostics? It would probably also be nice to have an option for the diagnostics handler to continuously populate the quickfix window as well, though that will probably not be very useful, since it will overwrite other usages of that window.

Edit: there'a already a function for setting the buffer diagnostics in a location list. So I'd like to have the equivalent function for setting the workspace diagnostics in a location list as well.

**Comment by tricktux** [comment]
Related to https://github.com/neovim/nvim-lspconfig/issues/69

**Comment by elianiva** [comment]
probably too overkill for what you want but there is [this plugin](https://github.com/folke/lsp-trouble.nvim) which I think is better than just a plain quickfix list.

**Comment by arsham** [comment]
I'm not sure if this is what you're after, but I've set these mappings to populate the local list and navigate through them:

```
nnoremap <buffer> <silent> <leader>dq <cmd>lua vim.lsp.diagnostic.set_loclist()<CR>
nnoremap <buffer> <silent> <leader>dn <cmd>lua vim.lsp.diagnostic.goto_next()<CR>
nnoremap <buffer> <silent> <leader>dp <cmd>lua vim.lsp.diagnostic.goto_prev()<CR>
```

**Comment by urandom** [comment]
Ah, so there's already a function for setting the diagnostic messages in the location list. I guess in the same light, we'd need a function to set all workspace diagnostics in a location list as well.

**Comment by mjlbach** [comment]
I'm not sure what this feature request is asking for, do you want `vim.lsp.diagnostic.set_loclist()` to have a toggle between buffer and workspace diagnostics? Do you want a `vim.lsp.diagnostic.set_quickfixlist()`?

**Comment by urandom** [comment]
@mjlbach 

at this point, for set_loclist to have an option to display the workspace diagnostics

**Comment by mjlbach** [comment]
Closed by #14511

**Comment by PhilippFeO** [comment]
> ```
> nnoremap <buffer> <silent> <leader>dq <cmd>lua vim.lsp.diagnostic.set_loclist()<CR>
> nnoremap <buffer> <silent> <leader>dn <cmd>lua vim.lsp.diagnostic.goto_next()<CR>
> nnoremap <buffer> <silent> <leader>dp <cmd>lua vim.lsp.diagnostic.goto_prev()<CR>
> ```

I just want to point out (for a future reader) that the afore mentioned functions are deprecated and won't work. The new functions are `vim.diagnostic.{setloclist, goto_next, goto_prev}()`, ie. `lsp` is removed.
